### PR TITLE
Feature/date picker

### PR DESCRIPTION
**Pull request by AndriuCoelho**
## Description
- Date Picker or Calendar is a date choice component. It is possible to limit a range of dates (useful for my bill payment team) or it is possible to initialize a common calendar.
Try to answer these questions:
- What is it about?
- Component of date choice
- How does it work?
It is possible to choose a date in a range or any date (depending on the initialization)
## Affected component
Is highly probable that this new code affects directly one (or more?) components inside this lib. Tell us who they are!
## RFC Link
- https://docs.google.com/document/d/1g31GzB9CqsGNrkGj6iDtjHBVVgGbfuNczIzTvah1dgQ/edit#heading=h.ixm8d2k39d9a
## Screenshots / GIFs

1) Calendar with range validation (useful for bill payment):

datepicker-gif2

2) Normal calendar:

datepicker-gif1

## Checks
Are you sure you followed all those steps? If so, repeat after me "I solemnly swear that ...":
   - [x] I have coded an example inside the Demo App,
   - [x] I have added proper tests,
   - [x] I have modified the Changelog.md file,
   - [ ] I have updated GitHub wiki,
   - [ ] I have the explicit approval of one or more members of the UX Team,
   - [x] I have checked that this code is ObjC compliant.
   - [ ] I have checked that all the new public enums conform to AndesEnumStringConvertible.
## Change type
This is easy, let us know what this code is about:
   - [x] This code adds a new component
   - [ ] This code includes improvements to an existent component
   - [ ] This code improves or modifies ONLY the Demo App.

### CODE REVIEW

File: LibraryComponents/Classes/Core/AndesDatePicker/Config/AndesDatePickerSettingCollection.swift [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    var currentDate: Date
+
+    weak var delegate: AndesDatePickerSettingCollectionDelegate?
+
+    private var calendar = Calendar(identifier: .iso8601)
+    private var header: AndesDatePickerHeaderView?
+
+    // MARK: - Initializer
+
+    init(baseDate: Date, daysToRender days: [AndesDayDatePicker]) {
```
Comment (joalonsopint) [inline LibraryComponents/Classes/Core/AndesDatePicker/Config/AndesDatePickerSettingCollection.swift:32]: Pienso que la variable days: [AndesDayDatePicker] no da claridad a que se está refiriendo, crees que podamos buscar un nombre que este un poco más asociado al objeto?

File: LibraryComponents/Classes/Core/AndesDatePicker/Model/AndesDayDatePicker.swift [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            } else {
+                selected = calendar.isDate(date, inSameDayAs: selectedDate ?? Date())
+            }
+
+            return AndesDayDatePicker(date: date, number: dateFormatter.string(from: date), selected: selected, isCurrentMonth: isCurrentMonth, endDate: dueDate, isValid: isValid)
+        }
+
+        days += generateStartOfNextMonth(using: firstDay, selectedDate: Date())
+
+        return days
```
Comment (joalonsopint) [inline LibraryComponents/Classes/Core/AndesDatePicker/Model/AndesDayDatePicker.swift:84]: Mismo comentario anterior, variable days

File: LibraryComponents/Classes/Core/AndesDatePicker/AndesDatePicker.swift [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    internal var contentView: AndesDatePickerView
+    private var datePickerView = AndesDatePickerDefaultView()
+    internal var didTapped: ((Date?, Bool) -> Void)?
+
+    /// We can initialize this component in two ways:
+    /// 1) Passing a deadline by parameter - we block the choice of the date until the date that was passed by parameter (useful for paying bills for example)
+    /// 2) Passing nothing by parameter - Works like a normal calendar
+
+    @objc public func setDatePickerDate(maxDate: Date? = nil, callback: @escaping ((Date?, Bool) -> Void)) {
```
Comment (joalonsopint) [inline LibraryComponents/Classes/Core/AndesDatePicker/AndesDatePicker.swift:22]: Te parece si añadimos documentación con las recomendaciones de Apple?, definiendo cada variable, así es fácil consultarla desde el mismo Xcode cuando se integra

File: LibraryComponents/Classes/Core/AndesDatePicker/Config/AndesDatePickerSettingCollection.swift [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+/// This class has all collectionview implementation (UICollectionViewDataSource and UICollectionViewDelegate)
+
+protocol AndesDatePickerSettingCollectionDelegate: AnyObject {
+    func didTouchNextMonth()
+    func didTouchPreviousMonth()
+    func didSelectDate(_ date: Date?)
+}
+
+@objc public class AndesDatePickerSettingCollection: NSObject, UICollectionViewDataSource {
```
Comment (joalonsopint) [inline LibraryComponents/Classes/Core/AndesDatePicker/Config/AndesDatePickerSettingCollection.swift:18]: Está bien que esta clase sea pública?

File: LibraryComponents/Classes/Core/AndesDatePicker/Model/AndesDayDatePicker.swift [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+//  Created by Ândriu Felipe Coelho on 10/11/20.
+//
+
+import Foundation
+
+protocol AndesDayDatePickerDelegate: class {
+    func didSelectEnabledDay(_ day: Date?)
+}
+
+@objc public class AndesDayDatePicker: NSObject {
```
Comment (joalonsopint) [inline LibraryComponents/Classes/Core/AndesDatePicker/Model/AndesDayDatePicker.swift:14]: Está bien que esta clase sea pública?

File: LibraryComponents/Classes/Core/AndesDatePicker/View/AndesDatePickerAbstractView.swift [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+    private func setup() {
+        loadNib()
+        datePickerView.layer.cornerRadius = 12
+        translatesAutoresizingMaskIntoConstraints = false
+        self.addSubview(datePickerView)
+        backgroundLayer = datePickerView.layer
+        datePickerView.pinToSuperview()
+
+        buttonPrimary.text = "Confirmar"
```
Comment (joalonsopint) [inline LibraryComponents/Classes/Core/AndesDatePicker/View/AndesDatePickerAbstractView.swift:88]: Esto podría estar localizado?

File: LibraryComponents/Classes/Core/AndesDatePicker/View/AndesDatePickerHeaderView.swift [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textAlignment = .center
+        label.textColor = UIColor.Andes.gray800
+        label.font = AndesStyleSheetManager.styleSheet.regularSystemFont(size: 16.0)
+
+        return label
+    }()
+
+    let previousButton: UIButton = {
```
Comment (joalonsopint) [inline LibraryComponents/Classes/Core/AndesDatePicker/View/AndesDatePickerHeaderView.swift:62]: Este podría ser privado como los demás?

**Review by joalonsopint — CHANGES_REQUESTED** [review]
Te dejé algunos comentarios

File: LibraryComponents/Classes/Core/AndesDatePicker/View/AndesDatePickerHeaderView.swift (hunk 7 above)
Comment (AndriuCoelho) [reply LibraryComponents/Classes/Core/AndesDatePicker/View/AndesDatePickerHeaderView.swift:62]: si

File: LibraryComponents/Classes/Core/AndesDatePicker/Config/AndesDatePickerSettingCollection.swift (hunk 1 above)
Comment (AndriuCoelho) [reply LibraryComponents/Classes/Core/AndesDatePicker/Config/AndesDatePickerSettingCollection.swift:32]: ok

File: LibraryComponents/Classes/Core/AndesDatePicker/Model/AndesDayDatePicker.swift (hunk 2 above)
Comment (AndriuCoelho) [reply LibraryComponents/Classes/Core/AndesDatePicker/Model/AndesDayDatePicker.swift:84]: ok

File: LibraryComponents/Classes/Core/AndesDatePicker/AndesDatePicker.swift (hunk 3 above)
Comment (AndriuCoelho) [reply LibraryComponents/Classes/Core/AndesDatePicker/AndesDatePicker.swift:22]: no entendí.. cual es la sugerencia?

Comment (joalonsopint) [reply LibraryComponents/Classes/Core/AndesDatePicker/AndesDatePicker.swift:22]: Añadir documentación, puedes hacerlo con el mismo Xcode automáticamente, te paso un link, https://nshipster.com/swift-documentation/
Si tienes dudas nos juntamos y lo vemos

Comment (AndriuCoelho) [reply LibraryComponents/Classes/Core/AndesDatePicker/AndesDatePicker.swift]: ok

File: LibraryComponents/Classes/Core/AndesTextField/View/TextArea/AndesTextAreaView.swift [hunk 8]
```diff
@@ -49,7 +49,7 @@ class AndesTextAreaView: AndesTextFieldAbstractView {
         textView.isScrollEnabled = false
         textView.clipsToBounds = false
 
-        textView.onNeedsBorderUpdate = { [weak self] (view: UIView) in
+        textView.onNeedsBorderUpdate = { [weak self] (_: UIView) in
```
Comment (joalonsopint) [inline LibraryComponents/Classes/Core/AndesTextField/View/TextArea/AndesTextAreaView.swift:52]: Esto cambió algo?

File: CHANGELOG.md [hunk 9]
```diff
@@ -19,6 +19,10 @@
 - AndesCheckbox lateral padding fixed  | Authors: [@ggiovanniniml](https://github.com/ggiovanniniml)
 
 
+# v3.15.0
```
Comment (joalonsopint) [inline CHANGELOG.md:22]: Creo que este debe ir en Unpublished

**Review by joalonsopint — APPROVED** [review]

**Pull request merged**